ibis/backends/oracle/compiler.py

```diff
@@ -0,0 +1,604 @@
+        else:
+            result = sg.select(STAR).from_(parent)
+            alias = None
+
+        if isinstance(n, int):
+            result = result.limit(n)
+        elif n is not None:
+            result = result.where(C.ROWNUM <= sg.select(n).from_(parent).subquery())
+        else:
+            assert n is None, n
+            if self.no_limit_value is not None:
+                result = result.limit(self.no_limit_value)
+
+        assert offset is not None, "offset is None"
+
+        if not isinstance(offset, int):
+            skip = offset
+            skip = sg.select(skip).from_(parent).subquery()
+        elif not offset:
+            if alias is not None:
+                return result.subquery(alias)
+            return result
+        else:
+            # add offset to where clause?
+            # but this assumes that there is a where clause
+            # assuming that gets handled, then
+            # next is to create a projection that sets ROWNUM as ora_rn, then
+            # add a predicate to that where ora_rn > offset
+            # result["where"] += offset
+            skip = offset
+
+        result = result.offset(skip)
+
+        if alias is not None:
+            return result.subquery(alias)
+        return result
+
+    @visit_node.register(ops.Date)
+    def visit_Date(self, op, *, arg):
+        return sg.cast(arg, to="date")
+
+    # @visit_node.register(ops.TimestampFromYMDHMS)
+    # def visit_TimestampFromYMDHMS(
+    #     self, op, *, year, month, day, hours, minutes, seconds
+    # ):
+    #     date_str = self.f.concat(
+    #         self.f.lpad(self.cast(year, dt.string), 4, "0"),
+    #         self.f.concat(
+    #             "-",
+    #             self.f.concat(
+    #                 self.f.lpad(self.cast(month, dt.string), 2, "0"),
+    #                 self.f.concat(
+    #                     "-",
+    #                     self.f.concat(
+    #                         self.f.lpad(self.cast(day, dt.string), 2, "0"),
+    #                         self.f.concat(
+    #                             "T",
+    #                             self.f.concat(
+    #                                 self.f.lpad(self.cast(hours, dt.string), 2, "0"),
+    #                                 self.f.concat(
+    #                                     ":",
+    #                                     self.f.concat(
+    #                                         self.f.lpad(
+    #                                             self.cast(minutes, dt.string), 2, "0"
+    #                                         ),
+    #                                         self.f.concat(
+    #                                             ":",
+    #                                             self.f.lpad(
+    #                                                 self.cast(seconds, dt.string),
+    #                                                 2,
+    #                                                 "0",
```

Comment:
Monstrous.

Reply:
It felt bad writing it and it feels bad looking at it